src/utils/errors.ts

```diff
@@ -6,3 +8,14 @@ export class UserRejectedRequestError extends Error {
     this.name = 'UserRejectedRequestError'
   }
 }
+
+export function throwReadableError(errorText: string, error: unknown) {
+  if (didUserReject(error)) {
```

Comment:
This should be independent of didUserReject to avoid confusing callsites. There are also cases (like calls to the routing api) where we want to throw the parsed error without checking for user rejection. What about just toReadableError, so the return value can be thrown?